### PR TITLE
fix flaubert tokenizer

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix an issue from #19330, see the comments in the changes.

Current test failure [here](https://github.com/huggingface/transformers/actions/runs/3231701081/jobs/5291558259)
```bash
E               if self.do_lowercase:
E           AttributeError: 'FlaubertTokenizer' object has no attribute 'do_lowercase'
```

### CODE REVIEW

File: src/transformers/models/flaubert/tokenization_flaubert.py [hunk 1]
```diff
@@ -260,11 +260,13 @@ def __init__(
             additional_special_tokens=additional_special_tokens,
             lang2id=lang2id,
             id2lang=id2lang,
-            do_lowercase_and_remove_accent=do_lowercase_and_remove_accent,
-            do_lowercase=do_lowercase,
             **kwargs,
         )
 
+        self.do_lowercase = do_lowercase
```
Comment (ydshieh) [inline src/transformers/models/flaubert/tokenization_flaubert.py:266]: This is never initialized in the parent classes, before/after PR #19330.

Before #19330, we needed to set it just like here.

File: src/transformers/models/flaubert/tokenization_flaubert.py [hunk 2]
```diff
@@ -260,11 +260,13 @@ def __init__(
             additional_special_tokens=additional_special_tokens,
             lang2id=lang2id,
             id2lang=id2lang,
-            do_lowercase_and_remove_accent=do_lowercase_and_remove_accent,
```
Comment (ydshieh) [inline src/transformers/models/flaubert/tokenization_flaubert.py:263]: Passing this made (just a bit) sense before #19330 when it was still a subclass of `XLMTokenizer`.
After that PR, we can just set it to `False` without passing to `super().__init__`.

File: src/transformers/models/flaubert/tokenization_flaubert.py [hunk 3]
```diff
@@ -260,11 +260,13 @@ def __init__(
...
-            do_lowercase=do_lowercase,
```
Comment (ydshieh) [inline src/transformers/models/flaubert/tokenization_flaubert.py:264]: This was never initialized in the parent classes  (`XLMTokenizer`, `PreTrainedTokenizer` or higher) before #19330, although it was passed to `XLMTokenizer`.

File: src/transformers/models/flaubert/tokenization_flaubert.py [hunk 4]
```diff
@@ -247,7 +247,7 @@ def __init__(
         ],
         lang2id=None,
         id2lang=None,
-        do_lowercase_and_remove_accent=True,
+        do_lowercase_and_remove_accent=True,  # Keep for backward-compatibility
```
Comment (sgugger) [inline src/transformers/models/flaubert/tokenization_flaubert.py:250]: Can we hide it in the kwargs, then in the init, pop it and send a user warning that it doesn't do anything if it's passed?

**Review by sgugger — COMMENTED** [review]
Thanks a lot for fixing this! Just left a comment on how to deal with the kwarg left for backward compatibility.

File: src/transformers/models/flaubert/tokenization_flaubert.py [hunk 5]
```diff
@@ -213,8 +213,6 @@ class FlaubertTokenizer(PreTrainedTokenizer):
             Dictionary mapping languages string identifiers to their IDs.
         id2lang (`Dict[int, str]`, *optional*):
             Dictionary mapping language IDs to their string identifiers.
-        do_lowercase_and_remove_accent (`bool`, *optional*, defaults to `True`):
-            Whether to lowercase and remove accents when tokenizing.
```
Comment (ydshieh) [inline src/transformers/models/flaubert/tokenization_flaubert.py:217]: I also removed this from the argument list in the doc. Hope this is a good decision :-)

Comment (sgugger) [reply src/transformers/models/flaubert/tokenization_flaubert.py:217]: Yes, thanks!

**Review by sgugger — APPROVED** [review]

**Pull request merged**